ui/src/main/java/org/pentaho/di/ui/spoon/delegates/SpoonClustersDelegate.java

```diff
@@ -85,27 +104,42 @@ private void showSaveError( ClusterSchema clusterSchema, KettleException e ) {
       getMessage( "Spoon.Dialog.ErrorSavingCluster.Message", clusterSchema.getName() ), e );
   }
 
-  public void editClusterSchema( TransMeta transMeta, ClusterSchema clusterSchema ) {
-    ClusterSchemaDialog dialog =
-        new ClusterSchemaDialog( spoon.getShell(), clusterSchema, transMeta.getClusterSchemas(), transMeta.getSlaveServers() );
-    if ( dialog.open() ) {
-      if ( spoon.rep != null && clusterSchema.getObjectId() != null ) {
-        try {
-          saveSharedObjectToRepository( clusterSchema, null );
-        } catch ( KettleException e ) {
-          showSaveError( clusterSchema, e );
+  public void editClusterSchema( ClusterSchemaManagementInterface manager, ClusterSchema clusterSchema ) {
+    String originalName = clusterSchema.getName().trim();
+    try {
+      // execution bowl for listing all slave servers
+      SlaveServerManagementInterface slaveServerManagementInterface =
+        spoon.getExecutionBowl().getManager( SlaveServerManagementInterface.class );
+
+      ClusterSchemaDialog dialog =
+          new ClusterSchemaDialog( spoon.getShell(), clusterSchema, manager.getAll(),
+                                   slaveServerManagementInterface.getAll() );
+      if ( dialog.open() ) {
+        String newName = clusterSchema.getName().trim();
+        manager.add( clusterSchema );
```

Comment:
Is there a check to make sure there is no deplicate before adding

Reply:
The ClusterSchemaDialog handles that.